x11-drivers/nvidia-drivers/nvidia-drivers-390.157-r1.ebuild

```
@@ -400,6 +400,13 @@ documentation that is installed alongside this README."
 	# manually if need others or addwrite)
 	insinto /etc/sandbox.d
 	newins - 20nvidia <<<'SANDBOX_PREDICT="/dev/nvidiactl"'
+
+	# Dracut does not include /etc/modprobe.d if hostonly=no, but we do need this
+	# to ensure that the nouveau blacklist is applied
+	# https://github.com/dracut-ng/dracut-ng/issues/674
+	# https://bugs.gentoo.org/932781
+	echo "install_items+=\" /etc/modprobe.d/nvidia.conf \"" >> \
+		"${ED}/usr/lib/dracut/dracut.conf.d/10-nvidia-drivers.conf" || die
```

Hm, I find appending to the linux-mod-r1 implementation-detail file rather than just doing a separate insinto+newins file a bit off, but guess having two nvidia files side-by-side may look weird to users so guess can keep it.

That aside, unlikely to be used on prefix but should still use the correct path, also minor style change to match the rest of the ebuild:
```suggestion
	echo "install_items+=\" ${EPREFIX}/etc/modprobe.d/nvidia.conf \"" >> \
		"${ED}"/usr/lib/dracut/dracut.conf.d/10-${PN}.conf" || die
```
Beside that lgtm, so feel free to merge (haven't tested w/ dracut, but I'm sure you did).

On a side-note, beside nouveau, this is also a minor security improvement in case users misuse the initramfs (may have already read https://bugs.gentoo.org/932781#c8).

---

Actually one more thing, revbumps on nvidia-drivers tend to disturb users and I'd rather not do another right away if possible.

It's also not super needed given nvidia-drivers is rebuilt with every kernel upgrades, so the file will propagate anyway.

---

> Actually one more thing, revbumps on nvidia-drivers tend to disturb users and I'd rather not do another right away if possible.
> 
> It's also not super needed given nvidia-drivers is rebuilt with every kernel upgrades, so the file will propagate anyway.

hrm.. then again, the file will be missing when the kernel builds the initramfs until the "next" upgrade... guess could keep them if you think it's important enough.

---

> hrm.. then again, the file will be missing when the kernel builds the initramfs until the "next" upgrade... guess could keep them if you think it's important enough.

And made me realize, guess concerning if the initramfs is built before nvidia-drivers is installed for the first time (aka nvidia.conf does not exist yet), meaning the very first build could load nouveau anyway. It'll also include an outdated copy in the event that nvidia-drivers changes it. But I see no solution to that other than making nvidia rebuild the initramfs unconditionally which would be terrible.

---

> hrm.. then again, the file will be missing when the kernel builds the initramfs until the "next" upgrade... guess could keep them if you think it's important enough.

I dropped the revbump, this is mostly for new users anyway. Existing nvidia-drivers users must have worked around this somehow, otherwise the driver would not work.

> And made me realize, guess concerning if the initramfs is built before nvidia-drivers is installed for the first time (aka nvidia.conf does not exist yet), meaning the very first build could load nouveau anyway. It'll also include an outdated copy in the event that nvidia-drivers changes it. But I see no solution to that other than making nvidia rebuild the initramfs unconditionally which would be terrible.

Aah good point, I think this is a problem we can "fix" by documenting it. I.e. after installing nvidia-drivers for the first time users should re-gen the initramfs.